ios/brave-ios/Sources/Brave/Frontend/Browser/BrowserViewController/BVC+WKNavigationDelegate.swift

```diff
@@ -682,11 +682,15 @@ extension BrowserViewController: WKNavigationDelegate {
       tab?.setCustomUserScript(scripts: scriptTypes)
     }
 
-    if let tab = tab,
-      let responseURL = responseURL,
-      InternalURL(responseURL)?.isSessionRestore == true
+    if let tab = tab, let responseURL = responseURL,
+      let response = response as? HTTPURLResponse
     {
-      tab.shouldNotifyAdsServiceTabDidChange = false
+      let isErrorPage = InternalURL(responseURL)?.isErrorPage == true || response.statusCode >= 400
+      let isSessionRestore = InternalURL(responseURL)?.isSessionRestore == true
```

Comment:
Rather than calling InternalUrl(responseURL)? twice, maybe assign to `url` and then in the condition call url.isErrorPage and url.isSessionRestore?

Reply:
Fixed